Source/WebCore/editing/Editor.cpp

```diff
@@ -1291,7 +1289,8 @@ void Editor::reappliedEditing(EditCommandComposition& composition)
 }
 
 Editor::Editor(Document& document)
-    : m_document(document)
+    : m_client(document.page() ? &document.page()->editorClient() : nullptr)
```

Comment:
Is it possible for `Editor::m_client` and `m_document.page()->editorClient()` to point to different (non-null) clients, in the case where a `Document` is detached from a frame and reattached to another frame?

Reply:
I don't believe so. The EditorClient is per Page, you would never expect a Document to get transferred from a page from another. As a matter of fact, Documents are generally not transferred from one frame to another either AFAIK. One exception may be the back/forward cache but in this case, the Page (and thus the EditorClient) would be the same when entering the back/forward cache and when exiting it.

If you'd like I can add an assertion to that effect.

Reply:
Sounds good! I do think an assertion would be helpful (since the invariant doesn't seem _super_ obvious)